transaction-status/src/lib.rs

```diff
@@ -241,6 +242,8 @@ pub struct Reward {
     pub lamports: i64,
     #[serde(deserialize_with = "default_on_eof")]
     pub post_balance: u64, // Account balance in lamports after `lamports` was applied
+    #[serde(default, deserialize_with = "default_on_eof")]
```

Comment:
`#[serde(deserialize_with = "default_on_eof")]` was insufficient to satisfy `serde_json` when trying to fetch an existing confirmed block from testnet without the `reward_type` field.  Adding `#[serde(default)]` as well made it happy

Reply:
With this PR, remove the `default` and then:
```
$ cargo run -- --verbose block 40700256
    Finished dev [unoptimized + debuginfo] target(s) in 0.34s
     Running `/Users/mvines/ws/s2/target/debug/solana --verbose block 40700256`
RPC URL: http://testnet.solana.com
Default Signer Path: /Users/mvines/.config/solana/id.json
Error: missing field `rewardType`
```

serde_json doesn't even hit the `default_on_eof` deserializer, it already decided that there's a missing field in the response and it can't do anything about it.  Deriving `Default` for the entire `Reward` struct produces the same error

Reply:
Yeah, `serde_json` doesn't need the `default_on_eof`. That's for bincode going into and out of blockstore.

...Ah, okay, figured it out. I was curious why this isn't affection `log_messages`, which also aren't on testnet right now. It's because the meta are being parsed by serde_json into `UiTransactionStatusMeta`, which don't include the `default_on_eof` tag.
Similarly, the serde_json parsing of `Reward` from rpc works fine with the `default_on_eof` tag removed. 

tl;dr, we'll need to remember `#[serde(default)]` for new `default_on_eof` fields on structs that are used directly in rpc.